Hangman/hangman.rb

```diff
@@ -0,0 +1,63 @@
+def contains?(character,key,current_word,guess_count,streak,score)
+  if key.include?(character)
+    i=0
+    if(current_word[key.index(character)]=="-")
+      while key[i..key.length].include?(character)
+        current_word[key[i..key.length].index(character)+i] = character
+        i = key[i..key.length].index(character)+i+1
+      end
+      streak += 1
+      score += streak*10
+    end
+    puts "Cool..."
+    return guess_count,score,streak
+  else
+    guess_count-=1
+    streak=0
+    puts "Oops!..."
+    return guess_count,score,streak
+  end
+end
+
+dictionary={
+  "WATER"=>"I am also known as life",
+  "PYTHON"=>"Programming language",
+  "APPLE"=>"I am a fruit and there is famous phrase on me",
+  "UBUNTU"=>"I am an OS. My name means Humanity",
+  "KALILINUX"=>"I am an OS. My name contains name of indian goddess",
+  "DEMOCRACY"=>"of the people, for the people, by the people",
+  "ANDROMEDA"=>"Galaxy name"
+  }
+score=0
+puts "-----Welcome to Hangman-----"
+dictionary.each do |key,val|
+  puts "Hint: => #{val}"
+  guess_count=3
+  current_word=""
+  streak=0
+  for i in 1..key.length do
+    current_word += "-"
```

Comment:
use single quote '-' / '-----Welcome to Hangman-----'  & dictionary  when you don't want to use ruby interpolation